forc-plugins/forc-client/src/op/call/call_function.rs

```diff
@@ -0,0 +1,748 @@
+        }
+        cmd::call::ExecutionMode::Simulate => {
+            forc_tracing::println_warning(&format!(
+                "Simulating transaction with wallet... {}",
+                wallet.address().hash()
+            ));
+            let tx = call
+                .with_external_contracts(external_contracts)
+                .transaction_builder(tx_policies, variable_output_policy, &wallet)
+                .await
+                .expect("Failed to build transaction")
+                .with_build_strategy(ScriptBuildStrategy::StateReadOnly)
+                .build(wallet.provider().unwrap())
+                .await?;
+            let tx_hash = tx.id(chain_id);
+            let gas_price = gas.map(|g| g.price).unwrap_or(Some(0));
+            let tx_status = wallet
+                .provider()
+                .unwrap()
+                .dry_run_opt(tx, false, gas_price)
+                .await
+                .expect("Failed to simulate transaction");
+            (tx_status, tx_hash)
+        }
+        cmd::call::ExecutionMode::Live => {
+            forc_tracing::println_action_green(
+                "Sending transaction with wallet",
+                &format!("0x{}", wallet.address().hash()),
+            );
+            let tx = call
+                .with_external_contracts(external_contracts)
+                .build_tx(tx_policies, variable_output_policy, &wallet)
+                .await
+                .expect("Failed to build transaction");
+            let tx_hash = tx.id(chain_id);
+            let tx_status = wallet
+                .provider()
+                .unwrap()
+                .send_transaction_and_await_commit(tx)
+                .await
+                .expect("Failed to send transaction");
+            (tx_status, tx_hash)
+        }
+    };
+
+    // Process transaction results
+    let receipts = tx_status
+        .take_receipts_checked(Some(&log_decoder))
+        .expect("Failed to take receipts");
+
+    // Parse the result based on output format
+    let mut receipt_parser = ReceiptParser::new(&receipts, DecoderConfig::default());
+    let result = match output {
+        cmd::call::OutputFormat::Default => {
+            let data = receipt_parser
+                .extract_contract_call_data(contract_id)
+                .expect("Failed to extract contract call data");
+            ABIDecoder::default()
+                .decode_as_debug_str(&output_param, data.as_slice())
+                .expect("Failed to decode as debug string")
+        }
+        cmd::call::OutputFormat::Raw => {
+            let token = receipt_parser
+                .parse_call(&Bech32ContractId::from(contract_id), &output_param)
+                .expect("Failed to extract contract call data");
+            token_to_string(&token).expect("Failed to convert token to string")
+        }
+    };
+
+    // Process and return the final output
+    let program_abi = sway_core::asm_generation::ProgramABI::Fuel(parsed_abi.clone());
```

Comment:
should be able to remove the clone here as `parsed_abi` isn't used again.

```suggestion
    let program_abi = sway_core::asm_generation::ProgramABI::Fuel(parsed_abi);
```

Reply:
https://github.com/FuelLabs/sway/pull/7056/commits/f592dfa74f031b9ae8ee95077f50157e619e69b9